recipes/jigsaw/build.sh

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env bash
+
+set -x
+set -e
+
+cd src
+
+${GXX} -std=c++11 -pedantic -Wall -s -O3 -flto -D NDEBUG -D __cmd_jigsaw \
+    -static-libstdc++ jigsaw.cpp -o jigsaw
+
+${GXX} -std=c++11 -pedantic -Wall -s -O3 -flto -D NDEBUG -D __cmd_tripod \
+    -static-libstdc++ jigsaw.cpp -o tripod
+
+${GXX} -std=c++11 -pedantic -Wall -O3 -flto -fPIC -D NDEBUG -D __lib_jigsaw \
+    -static-libstdc++ jigsaw.cpp -shared -o libjigsaw.so
+
+install -d ${PREFIX}/bin/
+for exec in jigsaw tripod
+do
+  install -m 755 ${exec} ${PREFIX}/bin/
+done
+
+install -d ${PREFIX}/lib/
+install -m 644 libjigsaw.so ${PREFIX}/lib/
+
+install -d ${PREFIX}/include/
+install -m 644 ../inc/*.h ${PREFIX}/include/
+
+# unit tests
+cd ../uni
+
+for test in 1 2 3 4 5
+do
+  ${GCC} -Wall test_${test}.c -Xlinker -rpath=${PREFIX}/lib -L ${PREFIX}/lib \
+      -ljigsaw -o test_${test}
+  ./test_${test}
```

Comment:
These unit tests build with a relative path to the include files.  Because of this, I am doing the tests here rather than during the "test" phase of the conda recipe.  There may be a way around this but my hope is that it is acceptable to do these tests during the "build" phase.

Reply:
It's nice to test if the relocation from the install prefix to testing prefix works.

You can persist files between building as testing using [this](https://docs.conda.io/projects/conda-build/en/latest/source/resources/define-metadata.html?highlight=selectors#source-files) and then pass `-I${PREFIX}/include`. (might need to be `${CONDA_PREFIX}`)

(It's not essential, just a suggestion for how you could do it.)

Reply:
Actually it looks like the testing is already fine. (And using `source_files`! 😆 )

Reply:
Thanks @chrisburr, these are a set of unit tests that differ a little bit from the tests done (with `source_files`.  I agree that it would be better to test with the include files once they're moved to the install location but I don't think doing these particular tests with the include files in `source_files` would really test that so I would prefer to leave it as is.